src/backend/logger/logger.ts

```diff
@@ -32,17 +23,42 @@ export enum LogPrefix {
   WineTricks = 'Winetricks'
 }
 
-let longestPrefix = 0
+type LogInputType = unknown[] | unknown
+
+// helper to convert LogInputType to string
+function convertInputToString(param: LogInputType): string {
+  const getString = (value: LogInputType): string => {
+    switch (typeof value) {
+      case 'string':
+        return value
+      case 'object':
+        // Object.prototype.toString.call(value).includes('Error') will catch all
+        // Error types (Error, EvalError, SyntaxError, ...)
+        if (Object.prototype.toString.call(value).includes('Error')) {
+          return value!.toString()
+        } else if (Object.prototype.toString.call(value).includes('Object')) {
+          return JSON.stringify(value, null, 2)
+        } else {
+          return `${value}`
+        }
+      case 'number':
+        return String(value)
+      case 'boolean':
+        return value ? 'true' : 'false'
+      default:
+        return `${value}`
+    }
+  }
 
-// helper to convert arrays to string
-function convertArrayToString(param: string | string[]): string {
-  if (typeof param === 'string') {
-    return param
-  } else if (Array.isArray(param)) {
-    return param.join(' ')
+  if (!Array.isArray(param)) {
+    return getString(param)
   }
-  // if somehow the param is an object and not array or string
-  return JSON.stringify(param)
+
+  const strings = [] as string[]
```

Comment:
This is probably nitpicky, but I'd recommend not getting used to using `as` too much
```suggestion
  const strings: string[] = []
```
does the same thing, but doesn't get you used to disabling the type checker :^)